### ISSUE TITLE
jitterFactor parameter does not behave as described

### ISSUE THREAD

**Issue by vigoo** [svroonland/rezilience#212]
I think the `jitterFactor` parameter of `Retry.Schedules.common` does not work as it's documentation and name suggests.

It says:

> Maximum fraction of the current delay interval that is randomly added or substracted. This helps to spread call attempts in time. To get a 10% jitter for example, use jitterFactor=0.1

From that I would expect that `jitterFactor = 0.0` completely disables jittering, while `jitterFactor = 1.0` uses the whole time range to calculate the jitter.

But it translates to `ZSchedule#jittered` with 

```scala
jittered(jitterFactor, 1.0 - jitterFactor)
```

which modifies the _delay_ to 

```scala
d * min * (1 - random) + d * max * random
```

where `d` is the original delay, `random` is between 0 and 1, `min` and `max` in our case is `jitterFactor` and `1.0 - jitterFactor`.

So for `jitterFactor = 0.0` it will be 

```scala
d * 0 * (1-random) + d * 1 * random == d * random
```

and for `jitterFactor = 1.0` it's 

```scala
d * 1 * (1-random) + d * 0 * random == d * (1 - random)
```

both cases basically resulting in the same. 

Based on this the value to turn jittering off would be `0.5` which results in `d * 0.5 * (1 - random) + d * 0.5 * random` that constantly halves the durations which is a bit unexpected.

I propose to explain the behavior in the docs and probably to have non-jittering variants

**Comment by svroonland** [comment]
Thanks for reporting @vigoo, I'll have a look soon.

**Comment by svroonland** [comment]
@vigoo What do you think of the fix in #221 ? That is how I intended it to work.

**Comment by vigoo** [comment]
Looks good!
